### ISSUE TITLE
[bug] Core ML export ignores softmax dim (wrong attr key → wrong axis)

### ISSUE THREAD

**Issue by ChanLumerico** [ChanLumerico/lucid#77]
**Location:** `lucid/coreml/_emit.py:308` (`_softmax`)

**Bug class:** wrong reduction axis (wrong trace-attribute key) → output normalized over the wrong dimension.

**Why it's wrong**
The emitter reads the softmax axis with the wrong key:
```python
axis = int(op.attrs.get("axis", -1))
```
but the C++ tracer records the softmax axis under the key `"dim"`, not `"axis"` (`lucid/_C/ops/ufunc/Softmax.cpp:58` and `:121` → `scope.set_attr("dim", ...)`; the MPSGraph emitter also reads `"dim"` at `lucid/_C/compile/OpEmitters/elementwise/Softmax.mm:24`). Since `"axis"` is never present, the `.get(..., -1)` default fires every time and the exported MIL `softmax` always reduces over the last axis. (Note the sibling emitters at lines 268 and 364 correctly read `"dim"`/`"keepdim"`, so this is a genuine key mismatch, not a naming convention.)

**Concrete failure case**
Export a model that calls `lucid.softmax(x, dim=1)` on a 4-D `(B, C, H, W)` tensor (e.g. channel-wise softmax in a segmentation head). The trace stores `dim=1`; the exported Core ML program softmaxes over axis `-1` (the `W` axis). The Core ML output is normalized over the wrong axis and will not match eager. Last-axis softmax (attention scores, classification logits) coincidentally still works because the ignored value equals the `-1` default, which is why existing transformer exports pass `verify()`.

**Suggested fix**
```python
axis = int(op.attrs.get("dim", -1))
```

**Comment by ChanLumerico** [comment]
Fixed the same way as #76: `_softmax` reads `dim`, through `_attr`,
which raises on a missing attribute rather than defaulting.

Checked just now with `lucid.softmax(x, dim=1)` on a `(1, 4, 3, 3)`
tensor — the shape where an axis mistake actually shows, since the
report is right that a last-axis softmax would have agreed either way:

```
max|diff| = 8.9e-08
```

"Only happened to be right because the models seen so far normalise the
last axis" is exactly what happened, and it is now written into the
`_attr` docstring so the next person meets the reasoning before the
bug.
